### DISCUSSION TITLE
Ignoring symbols

### DISCUSSION THREAD

**Discussion by WillsterJohnson** [colinhacks/zod#2679] [Q&A]
TL:DR;

How do I tell zod to completely ignore symbol keys in objects and arrays? I do not want to validate them, I want to invoke some `IF key IS symbol THEN ignore` routine.

---

I'm using [svelte](https://svelte.dev) and want to validate incoming json on the frontend. It's user data that stays in userland and frankly if a user wants to screw around in the app source, more power to them. Makes no difference to me in this situation.

I've found that I can get around the issue of reactivity and large data structures by blindly adding a store to everything `foo` where `foo !== null && typeof foo === "object"`. Inelegant, maybe, but it's effective and I no longer need to think about reactivity - it's always there.

I do this with an `ensureStores` function, then accessing the store with `getStore`. These functions have access to the symbol used to append the store to the object.

I would like Zod to ignore these symbols. In objects that's not gonna be too hard (though I'm open to recommendations), however for arrays, particularly tuples, this is going to (and already does) cause issues.

(using TS notation but I'm referring to Zod schemas) If I have some `[number, string, boolean]` and it gets given a `[number, string, [Symbol]: unknown]`, fair enough throw an error. But some `[number, string, boolean, [Symbol]: unknown]` ought to be seen as valid. This gets more complex in the case that the last one or more items are optional (`T | undefined`) .

Any ideas? I figured this would've worked as symbols are (to me at least) inherently private and not really "part" of the data structure.
